app/ante/ante.go

```diff
@@ -94,28 +97,48 @@ func NewAnteHandler(options ethante.HandlerOptions) (sdk.AnteHandler, error) {
 		// handle as totally normal Cosmos SDK tx
 		switch tx.(type) {
 		case sdk.Tx:
-			found := false
-			for _, msg := range tx.GetMsgs() {
-				switch msg.(type) {
-				// treat these three msg types differently because they might call EVM which results in massive gas consumption
-				// For these two msg types, we don't check gas limit by using a different ante handler
-				case *cctxtypes.MsgGasPriceVoter, *cctxtypes.MsgVoteOnObservedInboundTx:
-					found = true
-					break
-				case *stakingtypes.MsgCreateValidator:
-					if ctx.BlockHeight() == 0 {
-						found = true
-						break
-					}
+			anteHandler = newCosmosAnteHandler(options)
+			if len(tx.GetMsgs()) != 1 {
+				break
+			}
+
+			msg := tx.GetMsgs()[0] // now we must have len(tx.GetMsgs()) == 1
+			var innerMsg sdk.Msg
+			innerMsg = msg
+			if mm, ok := msg.(*authz.MsgExec); ok { // authz tx; look inside it
```

Comment:
Need more comment for details on this

Reply:
Does this mean only authz-wrapped system messages would be able to get reduced gas prices? 
If, for some reason, an operator wants to submit a TX manually using the operator key, the reduction won't be triggered.

Reply:
> Does this mean only authz-wrapped system messages would be able to get reduced gas prices? If, for some reason, an operator wants to submit a TX manually using the operator key, the reduction won't be triggered.

No, 
Look one line above: regardless of whether it's MsgExec wrapped, the innerMsg will point to the unwrapped Msg.